src/compiler/compiler.ml

```diff
@@ -1743,32 +1767,48 @@ end = struct
       let b = aux t in
       (* Format.eprintf "Bool is %b@." b; *)
       b
+    in *)
+    let is_eigen_variable ~depth = function
+      | Const b -> 0 <= b && b < depth
```

Comment:
This checks if the var is bound, not if it is the one bound by the pi.
```
main :-
  sigma x\
    foo x => foo _.
```
should be rejected

Reply:
It is already the case that `foo x` would be rejected, in fact, whenever I encounter a sigma in the term, I subst each occurences bound by it with a fresh uvar, therefore the only positive const are names https://github.com/LPCIC/elpi/blob/a2354c61821849644ff6a1eb3ceb14dbba863add/src/compiler/compiler.ml#L1804-L1807

Reply:
```
main :- pi x\ (bad :- (foo x) => foo x, bad) => bad.
```
Is this rejected? In the paper we do 1-pi, 1-=>, otherwise it is wrong.

Reply:
You are right, this should be rejected and it is not.
In this commit 7e1b6638c5b152a3c0e5b4e0d71fb1eb2f1056ad, I adress the problem by changing in which interval the id of the eigenvariable should be (i.e. I add a min_depth).
I put two tests, one with your code ([test111.elpi](https://github.com/LPCIC/elpi/blob/7e1b6638c5b152a3c0e5b4e0d71fb1eb2f1056ad/tests/sources/functionality/test111.elpi)) and one which is a slightly different version ([test112.elpi](https://github.com/LPCIC/elpi/blob/7e1b6638c5b152a3c0e5b4e0d71fb1eb2f1056ad/tests/sources/functionality/test112.elpi)).